### PR TITLE
feat(groups): Accept group in GraphQL for billable metric

### PR DESCRIPTION
**Pull request by rsempe**
## Roadmap Task

👉  https://github.com/getlago/lago/issues/32

## Context

We want to be able to price differently billable metrics according to several dimensions (groups).

For example, creating a `storage` pricing based on the `region of the server` and the `cloud provider` used.

## Description

The goal of this PR is to accept groups in GraphQL for a billable metric, on creation and update.

### CODE REVIEW

File: app/serializers/v1/billable_metric_serializer.rb [hunk 1]
```diff
@@ -10,7 +10,8 @@ def serialize
         description: model.description,
         aggregation_type: model.aggregation_type,
         created_at: model.created_at.iso8601,
-        field_name: model.field_name
+        field_name: model.field_name,
+        group: model.groups_as_tree
```
Comment (jdenquin) [inline app/serializers/v1/billable_metric_serializer.rb:14]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: spec/models/billable_metric_spec.rb [hunk 2]
```diff
@@ -28,4 +28,59 @@
...
+      before do
+        create(:group, billable_metric: billable_metric, key: 'country', value: 'france')
+        create(:group, billable_metric: billable_metric, key: 'country', value: 'italy')
+      end
+
+      it 'returns a tree with one dimension' do
+        expect(billable_metric.groups_as_tree).to eq(
+          {
+            key: 'country',
+            values: %w[france italy]
```
Comment (jdenquin) [inline spec/models/billable_metric_spec.rb:51]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: spec/models/billable_metric_spec.rb [hunk 3]
```diff
@@ -28,4 +28,59 @@
...
+
+      it 'returns a tree with two dimensions' do
+        expect(billable_metric.groups_as_tree).to eq(
+          {
+            key: 'country',
+            values: [
+              {
+                name: 'france',
+                key: 'cloud',
+                values: %w[aws google]
```
Comment (jdenquin) [inline spec/models/billable_metric_spec.rb:74]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: spec/models/billable_metric_spec.rb [hunk 4]
```diff
@@ -28,4 +28,59 @@
...
+              }, {
```
Comment (jdenquin) [inline spec/models/billable_metric_spec.rb:75]: Layout/MultilineArrayLineBreaks: Each item in a multi-line array must start on a separate line.

File: spec/models/billable_metric_spec.rb [hunk 5]
```diff
@@ -28,4 +28,59 @@
...
+                name: 'italy',
+                key: 'cloud',
+                values: %w[google]
```
Comment (jdenquin) [inline spec/models/billable_metric_spec.rb:78]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: spec/models/billable_metric_spec.rb [hunk 6]
```diff
@@ -28,4 +28,59 @@
...
+              }
+            ]
```
Comment (jdenquin) [inline spec/models/billable_metric_spec.rb:80]: Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.

File: spec/models/billable_metric_spec.rb [hunk 7]
```diff
@@ -28,4 +28,59 @@
...
+          }
```
Comment (jdenquin) [inline spec/models/billable_metric_spec.rb:81]: Style/TrailingCommaInArguments: Put a comma after the last parameter of a multiline method call. (https://rubystyle.guide#no-trailing-params-comma)

**Review by vincent-pochet — APPROVED** [review]

**Pull request merged**